File: library_common/src/main/java/com/phone/common_library/manager/GlideRoundManager.java
package com.phone.common_library.manager;

import android.graphics.drawable.Drawable;
import android.os.Build;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.target.CustomTarget;
import com.bumptech.glide.request.transition.Transition;
import com.phone.common_library.custom_view.GlideRoundTransform;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2021/3/31 10:55
 * introduce :
 */

public class GlideRoundManager {

    public static void setRoundCorner(final View view, final Drawable resourceId, final float cornerDipValue) {
        if (cornerDipValue == 0) {
            if (view.getMeasuredWidth() == 0 && view.getMeasuredHeight() == 0) {
                view.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
                    @Override
                    public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
                        view.removeOnLayoutChangeListener(this);
                        Glide.with(view)
                                .asDrawable()
                                .load(resourceId)
                                .transform(new CenterCrop())
                                .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                                .into(new CustomTarget<Drawable>() {
                                    @Override
                                    public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                        view.setBackground(resource);
                                    }

                                    @Override
                                    public void onLoadCleared(@Nullable Drawable placeholder) {

                                    }
                                });
                    }
                });
            } else {
                Glide.with(view)
                        .asDrawable()
                        .load(resourceId)
                        .transform(new CenterCrop())
                        .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                        .into(new CustomTarget<Drawable>() {
                            @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                            @Override
                            public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                view.setBackground(resource);
                            }

                            @Override
                            public void onLoadCleared(@Nullable Drawable placeholder) {

                            }
                        });
            }
        } else {
            if (view.getMeasuredWidth() == 0 && view.getMeasuredHeight() == 0) {
                view.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
                    @Override
                    public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
                        view.removeOnLayoutChangeListener(this);
                        Glide.with(view)
                                .load(resourceId)
                                .transform(new CenterCrop(), new RoundedCorners((int) cornerDipValue))
                                .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                                .into(new CustomTarget<Drawable>() {
                                    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                                    @Override
                                    public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                        view.setBackground(resource);
                                    }

                                    @Override
                                    public void onLoadCleared(@Nullable Drawable placeholder) {

                                    }
                                });
                    }
                });
            } else {
                Glide.with(view)
                        .load(resourceId)
                        .transform(new CenterCrop(), new RoundedCorners((int) cornerDipValue))
                        .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                        .into(new CustomTarget<Drawable>() {
                            @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                            @Override
                            public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                view.setBackground(resource);
                            }

                            @Override
                            public void onLoadCleared(@Nullable Drawable placeholder) {

                            }
                        });
            }
        }
    }

    public static void setCorners(final View view, final Drawable resourceId, final float leftTop_corner, final float leftBottom_corner, final float rightTop_corner, final float rightBottom_corner) {
        if (leftTop_corner == 0 && leftBottom_corner == 0 && rightTop_corner == 0 && rightBottom_corner == 0) {
            if (view.getMeasuredWidth() == 0 && view.getMeasuredHeight() == 0) {
                view.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
                    @Override
                    public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
                        view.removeOnLayoutChangeListener(this);
                        Glide.with(view)
                                .load(resourceId)
                                .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                                .into(new CustomTarget<Drawable>() {
                                    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                                    @Override
                                    public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                        view.setBackground(resource);
                                    }

                                    @Override
                                    public void onLoadCleared(@Nullable Drawable placeholder) {

                                    }
                                });
                    }
                });
            } else {
                Glide.with(view)
                        .load(resourceId)
                        .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                        .into(new CustomTarget<Drawable>() {
                            @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                            @Override
                            public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                view.setBackground(resource);
                            }

                            @Override
                            public void onLoadCleared(@Nullable Drawable placeholder) {

                            }
                        });
            }

        } else {
            if (view.getMeasuredWidth() == 0 && view.getMeasuredHeight() == 0) {
                view.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
                    @Override
                    public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
                        view.removeOnLayoutChangeListener(this);
                        GlideRoundTransform transform = new GlideRoundTransform(view.getContext(), leftTop_corner, leftBottom_corner, rightTop_corner, rightBottom_corner);
                        Glide.with(view)
                                .load(resourceId)
                                .transform(transform)
                                .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                                .into(new CustomTarget<Drawable>() {
                                    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                                    @Override
                                    public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                        view.setBackground(resource);
                                    }

                                    @Override
                                    public void onLoadCleared(@Nullable Drawable placeholder) {

                                    }
                                });
                    }
                });
            } else {
                GlideRoundTransform transform = new GlideRoundTransform(view.getContext(), leftTop_corner, leftBottom_corner, rightTop_corner, rightBottom_corner);
                Glide.with(view)
                        .load(resourceId)
                        .transform(transform)
                        .override(view.getMeasuredWidth(), view.getMeasuredHeight())
                        .into(new CustomTarget<Drawable>() {
                            @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
                            @Override
                            public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
                                view.setBackground(resource);
                            }

                            @Override
                            public void onLoadCleared(@Nullable Drawable placeholder) {

                            }
                        });
            }
        }
    }

}

File: library_common/src/main/java/com/phone/library_common/manager/GlideRoundManager.kt
package com.phone.library_common.manager

import android.graphics.drawable.Drawable
import android.os.Build
import android.view.View
import androidx.annotation.RequiresApi
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.target.CustomTarget
import com.bumptech.glide.request.transition.Transition
import com.phone.library_common.custom_view.GlideRoundTransform

object GlideRoundManager {

    @JvmStatic
    fun setRoundCorner(view: View, resourceId: Drawable?, cornerDipValue: Float) {
        if (cornerDipValue == 0f) {
            if (view.measuredWidth == 0 && view.measuredHeight == 0) {
                view.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
                    override fun onLayoutChange(
                        v: View,
                        left: Int,
                        top: Int,
                        right: Int,
                        bottom: Int,
                        oldLeft: Int,
                        oldTop: Int,
                        oldRight: Int,
                        oldBottom: Int
                    ) {
                        view.removeOnLayoutChangeListener(this)
                        Glide.with(view)
                            .asDrawable()
                            .load(resourceId)
                            .transform(CenterCrop())
                            .override(view.measuredWidth, view.measuredHeight)
                            .into(object : CustomTarget<Drawable?>() {
                                override fun onLoadCleared(placeholder: Drawable?) {}

                                override fun onResourceReady(
                                    resource: Drawable,
                                    transition: Transition<in Drawable?>?
                                ) {
                                    view.background = resource
                                }
                            })
                    }
                })
            } else {
                Glide.with(view)
                    .asDrawable()
                    .load(resourceId)
                    .transform(CenterCrop())
                    .override(view.measuredWidth, view.measuredHeight)
                    .into(object : CustomTarget<Drawable?>() {
                        override fun onLoadCleared(placeholder: Drawable?) {}

                        override fun onResourceReady(
                            resource: Drawable,
                            transition: Transition<in Drawable?>?
                        ) {
                            view.background = resource
                        }
                    })
            }
        } else {
            if (view.measuredWidth == 0 && view.measuredHeight == 0) {
                view.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
                    override fun onLayoutChange(
                        v: View,
                        left: Int,
                        top: Int,
                        right: Int,
                        bottom: Int,
                        oldLeft: Int,
                        oldTop: Int,
                        oldRight: Int,
                        oldBottom: Int
                    ) {
                        view.removeOnLayoutChangeListener(this)
                        Glide.with(view)
                            .load(resourceId)
                            .transform(CenterCrop(), RoundedCorners(cornerDipValue.toInt()))
                            .override(view.measuredWidth, view.measuredHeight)
                            .into(object : CustomTarget<Drawable?>() {
                                override fun onLoadCleared(placeholder: Drawable?) {}

                                override fun onResourceReady(
                                    resource: Drawable,
                                    transition: Transition<in Drawable?>?
                                ) {
                                    view.background = resource
                                }
                            })
                    }
                })
            } else {
                Glide.with(view)
                    .load(resourceId)
                    .transform(CenterCrop(), RoundedCorners(cornerDipValue.toInt()))
                    .override(view.measuredWidth, view.measuredHeight)
                    .into(object : CustomTarget<Drawable?>() {
                        override fun onLoadCleared(placeholder: Drawable?) {}

                        override fun onResourceReady(
                            resource: Drawable,
                            transition: Transition<in Drawable?>?
                        ) {
                            view.background = resource
                        }
                    })
            }
        }
    }

    @JvmStatic
    fun setCorners(
        view: View,
        resourceId: Drawable?,
        leftTop_corner: Float,
        leftBottom_corner: Float,
        rightTop_corner: Float,
        rightBottom_corner: Float
    ) {
        if (leftTop_corner == 0f && leftBottom_corner == 0f && rightTop_corner == 0f && rightBottom_corner == 0f) {
            if (view.measuredWidth == 0 && view.measuredHeight == 0) {
                view.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
                    override fun onLayoutChange(
                        v: View,
                        left: Int,
                        top: Int,
                        right: Int,
                        bottom: Int,
                        oldLeft: Int,
                        oldTop: Int,
                        oldRight: Int,
                        oldBottom: Int
                    ) {
                        view.removeOnLayoutChangeListener(this)
                        Glide.with(view)
                            .load(resourceId)
                            .override(view.measuredWidth, view.measuredHeight)
                            .into(object : CustomTarget<Drawable?>() {
                                override fun onLoadCleared(placeholder: Drawable?) {}

                                override fun onResourceReady(
                                    resource: Drawable,
                                    transition: Transition<in Drawable?>?
                                ) {
                                    view.background = resource
                                }
                            })
                    }
                })
            } else {
                Glide.with(view)
                    .load(resourceId)
                    .override(view.measuredWidth, view.measuredHeight)
                    .into(object : CustomTarget<Drawable?>() {
                        override fun onLoadCleared(placeholder: Drawable?) {}

                        override fun onResourceReady(
                            resource: Drawable,
                            transition: Transition<in Drawable?>?
                        ) {
                            view.background = resource
                        }
                    })
            }
        } else {
            if (view.measuredWidth == 0 && view.measuredHeight == 0) {
                view.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
                    override fun onLayoutChange(
                        v: View,
                        left: Int,
                        top: Int,
                        right: Int,
                        bottom: Int,
                        oldLeft: Int,
                        oldTop: Int,
                        oldRight: Int,
                        oldBottom: Int
                    ) {
                        view.removeOnLayoutChangeListener(this)
                        val transform = GlideRoundTransform(
                            view.context,
                            leftTop_corner,
                            leftBottom_corner,
                            rightTop_corner,
                            rightBottom_corner
                        )
                        Glide.with(view)
                            .load(resourceId)
                            .transform(transform)
                            .override(view.measuredWidth, view.measuredHeight)
                            .into(object : CustomTarget<Drawable?>() {
                                override fun onLoadCleared(placeholder: Drawable?) {}

                                override fun onResourceReady(
                                    resource: Drawable,
                                    transition: Transition<in Drawable?>?
                                ) {
                                    view.background = resource
                                }
                            })
                    }
                })
            } else {
                val transform = GlideRoundTransform(
                    view.context,
                    leftTop_corner,
                    leftBottom_corner,
                    rightTop_corner,
                    rightBottom_corner
                )
                Glide.with(view)
                    .load(resourceId)
                    .transform(transform)
                    .override(view.measuredWidth, view.measuredHeight)
                    .into(object : CustomTarget<Drawable?>() {
                        override fun onLoadCleared(placeholder: Drawable?) {}

                        override fun onResourceReady(
                            resource: Drawable,
                            transition: Transition<in Drawable?>?
                        ) {
                            view.background = resource
                        }
                    })
            }
        }
    }

}